### PR TITLE
Hardsuit helmet

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Refactor+balance change to hardsuit

Ports TG's refactor of hardsuit helmets, so they're integrated into the hardsuit by default; hardsuit helmets should never be used as a distinct item ever again that players can pick up themselves.

Refactor/Changes to the hardsuits

- Hardsuit helmets are part of the suit and can be toggled up/down with an action button
- Can no longer manually attach magboots to hardsuits

Balance change

- Hardsuits have an entry level melee resistance of 30, up from 10
- Increased security and HoS armor by 5
- HoS helmet armor matches his suit's armor
- increased the CE's hardsuit rad resist on the helmet by 10 (should now be fully rad proof with an engineering jumpsuit)
- Can't acid syndicate elite suits

:cl: Fox McCloud
tweak: Hardsuits have an integrated helmet; you no longer have to attach helmets
tweak: Toggling up/down helmets is now done with an action button
del: can no longer attach magboots to a hardsuit
tweak: Hardsuit base melee armor increased from 10 to 30
tweak: Security and HoS hardsuit melee armor increased by 5
tweak: HoS hardsuit helmet armor matches his hardsuit
tweak: CE's hardsuit helmet armor for rads increased by 10 (should be fully rad proof now)
tweak: Elite syndicate suits can't be acided
/:cl:

### CODE REVIEW

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -194,11 +194,7 @@
...
 	if(istype(W))
 		if(istype(W, /obj/item/clothing))
-			var/obj/item/clothing/C = W
-			if(C.hardsuit_restrict_helmet)
-				to_chat(src, "<span class='warning'>You must fasten the helmet to a hardsuit first. (Target the head and use on a hardsuit)</span>")// Stop eva helms equipping.
-			else
-				equip_to_slot_if_possible(C, slot)
+			equip_to_slot_if_possible(W, slot)
 		else
 			equip_to_slot_if_possible(W, slot)
```
Comment (Fox-McCloud) [inline code/modules/mob/mob.dm]: Change this; the istype clothing check isn't necessary anymore.

Comment (Shadow-Quill) [reply code/modules/mob/mob.dm]: >Reviews own PR 🤔

File: code/modules/clothing/suits/toggles.dm [hunk 2]
```diff
@@ -0,0 +1,76 @@
+//Hardsuit toggle code
+/obj/item/clothing/suit/space/hardsuit/New()
```
Comment (Couls) [inline code/modules/clothing/suits/toggles.dm]: shouldn't this use initialize?

Comment (Fox-McCloud) [reply code/modules/clothing/suits/toggles.dm]: The problem with `Initialize` is that we use `New` and `Initialize` both---and we can really only use `Initialize` if every single parent item also does.

Pretty sure other clothing (and items) don't use `Initialize` yet, sadly.

File: _maps/map_files/cyberiad/z2.dmm [hunk 3]
```diff
@@ -11644,8 +11639,8 @@
 /area/admin)
 "AL" = (
 /obj/structure/rack,
-/obj/item/storage/box/syndie_kit/hardsuit,
-/obj/item/storage/box/syndie_kit/elite_hardsuit{
+/obj/item/clothing/suit/space/hardsuit/syndi/elite,
```
Comment (Fox-McCloud) [inline _maps/map_files/cyberiad/z2.dmm]: Oh, one of them is regular.

File: code/modules/mining/equipment_locker.dm [hunk 4]
```diff
@@ -430,7 +430,7 @@
 		new /datum/data/mining_equipment("Silver Pickaxe",				/obj/item/pickaxe/silver,											1000),
 		new /datum/data/mining_equipment("Mining Conscription Kit",		/obj/item/storage/backpack/duffel/mining_conscript,					1500),
 		new /datum/data/mining_equipment("Jetpack",             		/obj/item/tank/jetpack/carbondioxide/mining,               			2000),
-		new /datum/data/mining_equipment("Mining Hardsuit",				/obj/item/storage/box/mininghardsuit,								2000),
+		new /datum/data/mining_equipment("Mining Hardsuit",				/obj/item/clothing/head/helmet/space/hardsuit/mining,				2000),
```
Comment (datlo) [inline code/modules/mining/equipment_locker.dm]: Shouldnt this give you the hardsuit? Unless im misunderstanding this the vendor will give you the helmet alone right now.

Comment (Fox-McCloud) [reply code/modules/mining/equipment_locker.dm]: oh, I made it give the wrong thing. Lol.

Comment (datlo) [reply code/modules/mining/equipment_locker.dm]: :+1:

**Comment by AffectedArc07** [comment]
Finally an end to people asking **"HOW DO I PUT A HARDSUIT HELMET ON?"**

**Comment by taukausanake** [comment]
Can we have prop versions of the helmets then? Special hats or something? I know it's really not needed but would be nice

**Pull request merged**